src/Profiles.ts

```diff
@@ -336,6 +347,224 @@ export class Profiles {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        const profileType = ZoweExplorerApiRegister.getMvsApi(deletedProfile).getProfileTypeName();
+        try {
+            this.deleteProf(deletedProfile, profileName, profileType);
+        } catch (error) {
+            this.log.error(localize("deleteProfile.delete.log.error", "Error encountered when deleting profile! ") + JSON.stringify(error));
+            await errorHandling(error, profileName, error.message);
+            throw error;
+        }
+
+        vscode.window.showInformationMessage("Profile " + profileName + " was deleted.");
+        return profileName;
+    }
+
+    public async deleteProf(ProfileInfo, ProfileName, ProfileType) {
+        let zosmfProfile: IProfile;
+        try {
+            zosmfProfile = await (await this.getCliProfileManager(ProfileType))
+            .delete({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+        } catch (error) {
+            vscode.window.showErrorMessage(error.message);
+        }
+        return zosmfProfile.profile;
+    }
+
+    public async deleteProfile(
+        datasetTree: IZoweTree<IZoweDatasetTreeNode>, ussTree: IZoweTree<IZoweUSSTreeNode>,
+        jobsProvider: IZoweTree<IZoweJobTreeNode>, node?: IZoweNodeType) {
+
+            let deleteLabel: string;
+            let deletedProfile: IProfileLoaded;
+            if (!node){
+                deletedProfile = await this.getDeleteProfile();
+                if (!deletedProfile) {
+                    return;
+                } else {
+                    deleteLabel = deletedProfile.name;
+                }
+            } else {
+                deletedProfile = node.getProfile();
+                deleteLabel = node.getProfileName();
+            }
+            const deleteSuccess = await this.deletePrompt(deletedProfile);
+            if (!deleteSuccess){
+                vscode.window.showInformationMessage(localize("deleteProfile.noSelected",
+                    "Operation Cancelled"));
+                return;
+            }
+
+            // Delete from Data Set Recall
+            const recallDs: string[] = datasetTree.getRecall();
+            for (let i = recallDs.length - 1; i >= 0; i--) {
+                const findNode = recallDs[i].substring(1, recallDs[i].indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    datasetTree.removeRecall(recallDs[i]);
+                }
+            }
+
+            // Delete from Data Set Favorites
+            const favoriteDs = datasetTree.mFavorites;
+            for (let i = favoriteDs.length - 1; i >= 0; i--) {
+                const findNode = favoriteDs[i].label.substring(1, favoriteDs[i].label.indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    await datasetTree.removeFavorite(favoriteDs[i]);
```

Comment:
🤔 Interesting, why `await` for `.removeFavorite`, but not for `.removeRecall`?

Reply:
I removed the await. I think it was a remnant from when i was debugging it